src/vs/workbench/api/node/extHostTerminalService.ts

```diff
@@ -23,6 +23,14 @@ export class ExtHostTerminalService extends BaseExtHostTerminalService {
 	public createTerminalFromOptions(options: vscode.TerminalOptions, internalOptions?: ITerminalInternalOptions): vscode.Terminal {
 		const terminal = new ExtHostTerminal(this._proxy, generateUuid(), options, options.name);
 		this._terminals.push(terminal);
+		if (options.location && typeof options.location === 'object' && 'parentTerminal' in options.location) {
+			const parentTerminal = options.location.parentTerminal;
+			if (parentTerminal) {
+				const parentExtHostTerminal = this._terminals.find(t => t.value === parentTerminal);
+				internalOptions = internalOptions ? internalOptions : {};
+				internalOptions.parentTerminal = parentExtHostTerminal;
```

Comment:
`internalOptions.location.parentTerminal`?

Reply:
not sure what you mean

Reply:
oh I see